crates/resolver/src/name_server_pool.rs

```diff
@@ -100,13 +107,28 @@ impl<P: ConnectionProvider> NameServerPool<P> {
                 next: AtomicUsize::new(0),
             }),
             active_requests: Arc::new(Mutex::new(HashMap::new())),
+            ttl: None,
         }
     }
 
     /// Returns the pool's options.
     pub fn context(&self) -> &Arc<PoolContext> {
         &self.state.cx
     }
+
+    /// Check if the TTL on the NameServerPool (if set) has expired
+    pub fn ttl_expired(&self) -> bool {
+        match self.ttl {
+            Some(ttl) => TtlInstant::now() > ttl,
+            None => false,
+        }
+    }
+
+    /// Set a TTL on the NameServerPool
+    pub fn with_ttl(mut self, ttl: Duration) -> Self {
+        self.ttl = Some(TtlInstant::now() + ttl);
+        self
+    }
```

Comment:
Nits: move these above `context()` (which is simpler/more boring) and reverse their order (keeping the builder method closer to the constructors).